src/vs/workbench/api/node/mainThreadHandlerRegistry.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+/**
+ * MainThreadHandlerRegistry maintains a collection of objects by handle.
+ */
```

Comment:
We don't _need_ it per se, but if we don't have it, then this logic gets inlined to $registerReferenceSupport and $provideReferenceProviderProgress in `src/vs/workbench/api/node/mainThreadLanguageFeatures.ts` (which makes those methods harder to read.

If any other methods add support for streaming/progress in the future, they will want similar logic so might as well make it reusable. This access protected access to the nested dictionary so the consumer doesn't need to worry about keys that don't exist.